lib/heimdallr/exquisite_corpse.rb

```diff
@@ -0,0 +1,230 @@
+  PERM_RW.can_send_messages = true
+
+  class ExquisiteCorpse
+    @@all = {}
+
+    def initialize(server, name)
+      @server = server
+      @name = name
+      @category = nil
+    end
+
+    def self.all(server)
+      @@all[server.id] = {} unless @@all.include? server.id
+
+      ecs = @@all[server.id]
+      categories = server.categories.filter { |cat| cat.name.start_with? 'ec-' }
+      names = categories.map { |cat| cat.name[3..] }
+      missing = names.filter { |name| !ecs.include? name }
+      missing.each { |name| ecs[name] = ExquisiteCorpse.new server, name }
+      ecs
+    end
+
+    def as_member(user)
+      if user.instance_of? Discordrb::Member
+        member
+      elsif user.instance_of? Discordrb::User
+        @server.member(user.id)
+      elsif user.instance_of? String
+        @server.member Integer(user.tr('<>@!', '    ').strip)
+      else
+        raise TypeError, "Member #{user} of unknown type #{user.class}."
+      end
+    end
+    private :as_member
+
+    def create_main_channels(
+      channel_category, created_participant_role, created_viewer_role, reason
+    )
+      everyone = everyone_role
+      @server.create_channel(
+        'ec-general',
+        topic: 'Meta discussions about the exquisite corpse. No spoilers!',
+        parent: channel_category,
+        reason: reason,
+        permission_overwrites: [
+          Discordrb::Overwrite.new(everyone, deny: PERM_RW),
+          Discordrb::Overwrite.new(created_participant_role, allow: PERM_RW)
+        ]
+      )
+      @server.create_channel(
+        'ec-result',
+        topic: 'The resulting story.',
+        parent: channel_category,
+        reason: reason,
+        permission_overwrites: [
+          Discordrb::Overwrite.new(everyone, deny: PERM_RW),
+          Discordrb::Overwrite.new(created_viewer_role, allow: PERM_R)
+        ]
+      )
+    end
+    private :create_main_channels
+
+    def create_writing_channels(channel_category, participants, reason)
+      everyone = everyone_role
+      prev = participants.last
+      participants.each do |participant|
+        @server.create_channel(
+          "#{prev.display_name} -> #{participant.display_name}",
+          topic: "Story bits written by #{prev.display_name}.",
+          parent: channel_category,
+          reason: reason,
+          permission_overwrites: [
+            Discordrb::Overwrite.new(everyone, deny: PERM_RW),
+            Discordrb::Overwrite.new(prev, allow: PERM_RW),
+            Discordrb::Overwrite.new(participant, allow: PERM_R)
+          ]
+        )
+        prev = participant
+      end
+    end
+    private :create_writing_channels
+
+    def category
+      if @category.nil?
+        categories = @server.categories.filter do |cat|
+          cat.name == "ec-#{@name}"
+        end
+        unless categories.one?
+          names = @server.categories.map(&:name)
+          raise "Exquisit Corpse's channel category 'ec-#{@name}' not found in #{names}."
```

Comment:
```suggestion
          raise "Exquisite Corpse's channel category 'ec-#{@name}' not found in #{names}."
```
Same for the other exceptions